src/views/Main/index.jsx

```diff
@@ -174,61 +183,69 @@ class MainContainer extends Component {
       event.preventDefault();
       const { partialOrg } = this.state;
       this.setState({
-        showPerson: partialOrg[properties.ldapEmail],
+        personDetails: partialOrg[properties.ldapEmail],
       });
     }
 
     handleComponentBackToMenu(event) {
       event.preventDefault();
       this.setState({
-        showComponent: undefined,
-        showPerson: undefined,
+        componentDetails: undefined,
+        personDetails: undefined,
       });
     }
 
     render() {
       const {
-        showComponent,
-        showPerson,
+        componentDetails,
+        personDetails,
         bugzillaComponents,
         ldapEmail,
         partialOrg,
         teamComponents,
-        selectedTabIndex,
       } = this.state;
       const { context } = this;
       const userSession = context.getUserSession();
+      const { match } = this.props;
+      const view = match.params.view || 'reportees';
 
       return (
         <div>
           <Header
-            selectedTabIndex={selectedTabIndex}
-            handleTabChange={this.handleChangeSelectedTab}
+            selectedTabIndex={VIEW_TO_TAB_INDEX[view]}
+            handleTabChange={this.handleNavigateAndClear}
           />
           {!userSession && <h3>Please sign in</h3>}
-          {showComponent && (
+          {componentDetails && (
             <BugzillaComponentDetails
-              {...showComponent}
-              title={showComponent.title}
+              {...componentDetails}
               onGoBack={this.handleComponentBackToMenu}
             />
           )}
-          {showPerson && (
+          {personDetails && (
             <PersonDetails
-              person={showPerson}
+              person={personDetails}
               bugzillaComponents={Object.values(bugzillaComponents)}
               onGoBack={this.handleComponentBackToMenu}
             />
           )}
-          {!showComponent && !showPerson && partialOrg && userSession && (
-            <MainView
+          {view === 'reportees' && partialOrg && (
+            <Reportees
               ldapEmail={ldapEmail}
               partialOrg={partialOrg}
+              onPersonDetails={this.handleShowPersonDetails}
+            />
+          )}
+          {view === 'teams' && (
+            <BugzillaComponents
+              bugzillaComponents={Object.values(teamComponents)}
+              onComponentDetails={this.handleShowComponentDetails}
+            />
+          )}
+          {view === 'components' && partialOrg && (
```

Comment:
You can avoid

```jsx
{view === ... && ()}
{view === ... && ()}
{view === ... && ()}
...
```

by doing https://github.com/mozilla/bugzilla-dashboard/pull/60#discussion_r273473508.

I also recommend making use of lazy loading by using [`React.lazy`](https://reactjs.org/docs/code-splitting.html#reactlazy) otherwise all of the components will be downloaded on load time which will affect the initial loading time in the future as the application grows.

Reply:
Thanks for pointing out React.lazy!

Let's move the rest of the conversation to the other one.